### DISCUSSION TITLE
Dynamically configure validation groups on `MapRequestPayload` and `MapQueryString` attributes

### DISCUSSION THREAD

**Discussion by milanqtx** [symfony/symfony#51012] [Q&A] [answered]
Hi everyone - Is it possible to dynamically configure validation groups on the new `MapRequestPayload` and `MapQueryString` and attributes?
_Something similar to the group resolves for forms https://symfony.com/doc/current/form/validation_group_service_resolver.html_

I was imagining something like this:
```php
    public function __invoke(
        #[MapQueryString(validationGroups: OrdersQueryDtoGroupResolver::class)] OrdersQueryDto $query,
    ): Response {
        // ...
    }
```

**Comment by HypeMC** [comment]
Hi, no currently this is not supported.

**Comment by yceruto** [comment, marked as answer]
If the validation groups depend on the state of the underlying object you can use `#[Assert\GroupSequenceProvider]` and determine the groups dynamically.
https://symfony.com/doc/current/validation/sequence_provider.html#group-sequence-providers

**Reply by milanqtx** [reply]
Thank you for your help! 🙏🏻
